src/main/java/com/denizenscript/depenizen/bukkit/events/residence/ResidenceGetsDeletedScriptEvent.java

```diff
@@ -0,0 +1,72 @@
+package com.denizenscript.depenizen.bukkit.events.residence;
+
+import com.bekvon.bukkit.residence.event.ResidenceDeleteEvent;
+import com.denizenscript.denizen.events.BukkitScriptEvent;
+import com.denizenscript.denizen.utilities.implementation.BukkitScriptEntryData;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import com.denizenscript.denizencore.scripts.ScriptEntryData;
+import com.denizenscript.depenizen.bukkit.objects.residence.ResidenceTag;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+
+public class ResidenceGetsDeletedScriptEvent extends BukkitScriptEvent implements Listener {
+
+    // <--[event]
+    // @Events
+    // residence gets deleted
+    //
+    // @Switch cause:<cause> to only process the event if the cause matches specified cause.
+    //
+    // @Triggers when a Residence gets deleted.
+    //
+    // @Context
+    // <context.cause> Returns the cause of deletion. ( Available causes: PLAYER_DELETE, OTHER, LEASE_EXPIRE )
```

Comment:
Should match the format used by other events, something like `...deletion. Can be: X, Y, or Z`